server.go

```diff
@@ -72,9 +72,10 @@ type response struct {
 	tsigStatus     error
 	tsigRequestMAC string
 	tsigSecret     map[string]string // the tsig secrets
-	udp            *net.UDPConn      // i/o connection if UDP was used
+	udp            net.PacketConn    // i/o connection if UDP was used
 	tcp            net.Conn          // i/o connection if TCP was used
 	udpSession     *SessionUDP       // oob data to get egress interface right
+	pcSession      net.Addr          // address to use when writing to a generic net.PacketConn
```

Comment:
pcSession needs to be carried separately from the PacketConn?

Reply:
This gives us the peers remote address, we can only get our local address from the connection because it’s an unconnected listen socket.